AVsitter2/Plugins/AVprop/[AV]menu.lsl

```diff
@@ -627,6 +628,17 @@ default
                 {
                     MTYPE = (integer)part0;
                 }
+				else if (command == "LMSOURCE")
+				{
+					if(llToLower(part0) == "self")
+					{
+						LMSOURCE = -1;
```

Comment:
These codes are strange.  It's preferable to start numbering at 0 and go up, especially if they are going to be numeric codes in the notecard, where 0 would be the traditional/default behaviour and 1 the first non-default. This would also mean that it can be treated as boolean while there are only two possibilities, and change if the need arises.

Reply:
Hello Sei-Lisa, this is Happ. I'm collaborating with Syn on the contribution, and she's asked me to offer an explanation for two of the design decisions we opted for: keywords instead of numbers, and use of negative numbers in the internal tokens.

* Internally we chose negative values in order to leave room in case anyone feels like supporting explicit link numbers in future, since the latter would be potentially any positive number between 1-255 and in the degenerate case even 0 would be a valid link number.

* keywords instead of numbers: from our perspective as users of the product, we rate the reliance of numbers in MTYPE as relatively user unfriendly. They're a challenge to keep memorized, so you have to look them up every time you establish a new MTYPE or when reviewing previous products to double check what effect the notecard setting might have had. Keywords are simply self-documenting.

As far as memory goes, we are still representing the value internally as an integer and the variable which reads this value out of the notecard is already a string which gets garbage collected as soon as dataserver finishes processing that line of the notecard. So I am curious which surface of the memory profile has you concerned?

We'll get an update to our submission that covers the remaining changes as soon as time allows, but let us know your thoughts about these two points so we can figure out the best way to proceed. Thank you, Sei-Lisa. :)

Reply:
Can you give an example of a use case where only messages that come from a specific prim are accepted?

Reply:
Not off the top of my head, no. I just did my reflexive brainstorming for user input fields of "what kinds of values are within the realm of comprehension that somebody might demand down the road that it would save me time to make sufficient room for today?" And the first thing that came to mind was the same idea as link-specification in LSL where you can input a specific number 0+ or input a constant (eg, keyword) that maps to a negative value internally.

That said (and I'm embarrassed not to have thought about it before), how would you feel if we just dropped the other shoe and used the LSL constants directly instead of magic numbers? LINK_SET and LINK_THIS?

Reply:
Since it's a pretty advanced setting that will be very seldom used, I think that the memory argument has more weight than the user-friendliness argument. Just calling a function (llToLower in this case) involves quite some overhead, and the strings take 2 bytes per character plus 1 byte for a null terminator plus 1 byte or more for the length, in addition to the 5 bytes needed to reference them. And given that neither of us can see any uses for using link numbers as possible values of the setting, let's do it as I suggested. If someone needs to filter out all prims but a specific one, the source is available for them to adapt it to their needs.

Amend and force-push would be preferred in this case, since it's a short patch easy to review.

Reply:
Roger that, thanks for hearing me out. :)